spikeinterface/sortingcomponents/dimensionality_reduction.py

```diff
@@ -0,0 +1,99 @@
+import pickle
+
+import numpy as np
+
+from spikeinterface.sortingcomponents.peak_pipeline import run_peak_pipeline, PeakPipelineStep
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+from spikeinterface.sortingcomponents.peak_selection import select_peaks
+from spikeinterface.postprocessing import compute_principal_components
+from spikeinterface.core.template_tools import get_template_channel_sparsity
+from spikeinterface import extract_waveforms
+from spikeinterface import NumpySorting
+
+
+class WaveformStep(PeakPipelineStep):
+    need_waveforms = True
+
+    def __init__(self, recording, ms_before=1., ms_after=1.,  peak_sign='neg', all_channels=True):
+        PeakPipelineStep.__init__(self, recording, ms_before=ms_before, ms_after=ms_after)
+        self.all_channels = all_channels
+        self.peak_sign = peak_sign
+        self._kwargs.update(dict(all_channels=all_channels, peak_sign=peak_sign))
+        self._dtype = recording.get_dtype()
+
+    def get_dtype(self):
+        return self._dtype
```

Comment:
This could be moved to `PeakPipelineStep` to optionally return waveforms with peaks

Reply:
I removed this from here and did this in another PR #1182